pages/_document.tsx

```diff
@@ -127,7 +128,8 @@ class CustomDocument extends Document {
     const isProd = process.env.NODE_ENV === 'production';
     const useReact =
       this.props['__NEXT_DATA__']?.props?.pageProps?.metadata?.useReact;
-
+    const session =
+      this.props['__NEXT_DATA__']?.props?.pageProps?.metadata?.session;
```

Comment:
I see an issue here : 
props get serialized by next but session is supposed to be crypted is nt it a big security breach to serialize the session into the props ? 

Could we maybe use an ENUM for the different agent levels or use another way to inform the `_document` that the session is agent/super-agent without serializing the entire session object ?

Reply:
We already serialize the whole session object in the `NEXT_DATA` props for every page with react enabled. The change here doesn't modify the session object, nor the surface of information accessible from the front, only read in it. 

Unless there is something I missed?